### PR TITLE
feat(plugins): adds opsgenie plugin

### PR DESCRIPTION
**Pull request by scefali**
Must be merged with: https://github.com/getsentry/getsentry/pull/3374

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -159,8 +160,9 @@ def run(self):
             "heroku = sentry_plugins.heroku.plugin:HerokuPlugin",
             "jira = sentry_plugins.jira.plugin:JiraPlugin",
             "jira_ac = sentry_plugins.jira_ac.plugin:JiraACPlugin",
-            "phabricator = sentry_plugins.phabricator.plugin:PhabricatorPlugin",
+            "opsgenie = sentry_plugins.opsgenie.plugin:OpsGeniePlugin",
             "pagerduty = sentry_plugins.pagerduty.plugin:PagerDutyPlugin",
+            "phabricator = sentry_plugins.phabricator.plugin:PhabricatorPlugin",
```
Comment (scefali) [inline setup.py]: I realized that the `phabricator` should be after `pagerduty` so I made that change as well

File: src/sentry_plugins/opsgenie/plugin.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                "Logger": group.logger,
+                "Level": group.get_level_display(),
+                "URL": group.get_absolute_url(),
+                "Triggering Rules": json.dumps(triggering_rules),
+            },
+            "entity": group.culprit,
+        }
+
+        payload["tags"] = [
+            "%s:%s" % (str(x).replace(",", ""), str(y).replace(",", ""))
```
Comment (scefali) [inline src/sentry_plugins/opsgenie/plugin.py]: Do I need to use `six.binary_type` here?

Comment (wedamija) [reply src/sentry_plugins/opsgenie/plugin.py]: `six.text_type`

**Review by wedamija — APPROVED** [review]

File: src/sentry_plugins/opsgenie/plugin.py (hunk 2 above)
Comment (joshuarli) [reply src/sentry_plugins/opsgenie/plugin.py]: ^

Comment (scefali) [reply src/sentry_plugins/opsgenie/plugin.py]: @joshuarli I see the linter telling me to use binary_type in other cases. How do I know which one to use?

**Review by joshuarli — APPROVED** [review]
lgtm after you change over to `six.text_type`

**Pull request merged**